homeassistant/components/fireservicerota/binary_sensor.py

```diff
@@ -0,0 +1,103 @@
+"""Binary Sensor platform for FireServiceRota integration."""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DOMAIN as FIRESERVICEROTA_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up FireServiceRota binary sensor based on a config entry."""
+    coordinator = hass.data[FIRESERVICEROTA_DOMAIN][entry.entry_id]
+
+    async_add_entities([ResponseBinarySensor(coordinator, entry)], True)
+
+
+class ResponseBinarySensor(BinarySensorEntity, CoordinatorEntity):
+    """Representation of an FireServiceRota sensor."""
+
+    def __init__(self, coordinator: DataUpdateCoordinator, entry):
+        """Initialize."""
+        self._coordinator = coordinator
+        super().__init__(coordinator)
+        self._unique_id = entry.unique_id
+
+        self._state = None
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return "Duty"
+
+    @property
+    def icon(self) -> str:
+        """Return the icon to use in the frontend."""
+        return "mdi:calendar"
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this binary sensor."""
+        return f"{self._unique_id}_Duty"
```

Comment:
Ah, you do that here. Just do it in constructor.

Reply:
Done